modules.d/35network-legacy/net-genrules.sh

```diff
@@ -96,7 +96,11 @@ command -v fix_bootif >/dev/null || . /lib/net-lib.sh
 
         for iface in $IFACES; do
             if [ "$bootdev" = "$iface" ] || [ "$NEEDNET" = "1" ]; then
-                echo "${DRACUT_SYSTEMD+systemctl is-active initrd-root-device.target || }[ -f /tmp/net.${iface}.did-setup ]" >$hookdir/initqueue/finished/wait-$iface.sh
+		if [ -n "$netroot" ] && [ -n "$DRACUT_SYSTEMD" ]; then
+                    echo "systemctl is-active initrd-root-device.target || [ -f /tmp/net.${iface}.did-setup ]"
```

Comment:
Hmm, NEEDNET=1 means, that somebody requested to wait until all network interfaces are up, before continuing.
This patch will change this behavior!

Reply:
`95iscsi/module-setup.sh` sets "`rd.neednet`" unconditionally (which is what's causing us trouble here). So does `35network-manager/nm-config.sh`. `35network-legacy/parse-ip-opts.sh` sets it if more then one `ip=` parameter is found and `rd.neednet` isn't explicitly disabled.

So the actual meaning of "`rd.neednet`" seems to be "**some** network interface needs to be up", rather than "**all** network interfaces must be up". 

The rationale of this patch is "*if it's about booting a computer* (indicated by `$netroot`), availability of the root device is sufficient to continue".  Can you give examples where this assumption is false, and where this patch would thus cause a regression?

If it's some other use of dracut (other than booting a computer), `$netroot` will not be set and dracut will wait for all interfaces as usual.

The root of the issue is that AFAICS, dracut does not have a concept of multipathing. It happens to work with Fibre Channel, because the connections don't have to be brought up actively. But with (software) iSCSI, it seems to be impossible to tell dracut "any one of these interfaces / iSCSI connections would do". Rather, dracut tries to bring up every interface and every iSCSI connection configured, and waits for each one, which contradicts the spirit of multipathing. This is what I'm trying to fix.

In the long run, a more generalized new approach to iSCSI multipath boot in dracut would be desirable, but that would be a larger effort.

Reply:
If I'm overlooking some simple way to express the "or" logic for network interface bringup logic in dracut, I'd be eager to learn about it.

Reply:
Note: the man page says

     rd.neednet=1
            boolean, bring up network even without netroot set

So even there, it doesn't say "bring up all network interfaces"; rather, it hints that it's a setting that's (only?) relevant without netroot (where my patch doesn't change behavior).

Anyway, I'm aware that @haraldh, your interpretation counts more than anyone else's.
Please clarify if this is a killer argument, and if possible, give us a hint how to fix it.

One option would be to fix the cases where `rd.neednet` is set without compellent reason (see above), and modify my patch such that it wouldn't do anything with `$NEEDNET = 1`. Would that be ok for you?

Reply:
Oh, totally misread that patch somehow... sorry.. seems to be fine with me.